### PR TITLE
:seedling: Update mdbook to v0.4.3

### LINKED ISSUE #3658: Book preview does not render tabs
**What steps did you take and what happened:**
If I open the preview of the Cluster API book at https://master.cluster-api.sigs.k8s.io/user/quick-start.html is does not see tabs:
Screenshot 2020-09-17 at 14 53 10

**What did you expect to happen:**
To see tabs, like I can see when I run `make serve-book`:
Screenshot 2020-09-17 at 14 50 19

**Anything else you would like to add:**
From a quick look at the HTML, the web site at master.cluster-api.sigs.k8s.io does not apply transformations to {{tabs}}{{tab}} directives at all, so most probably there is some problem in the site generation part, not in the page source. 

/kind bug
/area documentation

### PR DESCRIPTION
**Pull request by vincepri**
Signed-off-by: Vince Prignano <vincepri@vmware.com>

**What this PR does / why we need it**:
This PR updates mdbook, which is currently using a really old version.

Fixes #3658 

/milestone v0.3.10
/assign @fabriziopandini @ncdc

### CODE REVIEW

**Comment by vincepri** [comment]
This fixes the rendering issue https://5f6377f19d7cbf0007545a88--kubernetes-sigs-cluster-api.netlify.app/user/quick-start.html

File: docs/book/install-and-build.sh [hunk 1]
```diff
@@ -54,9 +54,9 @@ esac
 
 # grab mdbook
 # we hardcode linux/amd64 since rust uses a different naming scheme and it's a pain to tran
-echo "downloading mdBook-v0.3.1-${arch}-${target}.${ext}"
+echo "downloading mdBook-v0.4.3-${arch}-${target}.${ext}"
```
Comment (ncdc) [inline docs/book/install-and-build.sh:57]: Want to pull the version into a var and reuse it?

Comment (vincepri) [reply docs/book/install-and-build.sh]: Done!

**Pull request merged**